restapi/src/main/java/io/stargate/web/resources/RowResource.java

```diff
@@ -400,12 +400,12 @@ private static Object filterToValue(Object val, String column, Table tableData)
   private List<Where<?>> buildWhereClause(PathSegment id, Table tableMetadata) {
     List<String> values = idFromPath(id);
 
-    final List<Column> keys = tableMetadata.partitionKeyColumns();
-    if (keys.size() != values.size()) {
-      throw new IllegalArgumentException("not enough partition keys provided");
+    final List<Column> keys = tableMetadata.primaryKeyColumns();
+    if (keys.size() < values.size()) {
+      throw new IllegalArgumentException("not enough keys provided");
```

Comment:
Maybe we can place here a message that will be more helpful from user perspective?
```
Invalid number of key values required (N).
All partition key columns values are required plus 0..all clustering columns values in proper order.
```

Reply:
Good call.